### PR TITLE
added Course Form

### LINKED ISSUE #24: Add form for creating a course
Add a form to the client that allows users to create courses

### PR DESCRIPTION
**Pull request by kevinzhong930**
Created a form to add courses.

Resolves #24

### CODE REVIEW

File: devU-client/src/components/pages/userCoursesListPage.tsx [hunk 1]
```diff
@@ -68,10 +74,18 @@ const UserCoursesListPage = () => {
...
   return (
     <PageWrapper>
       <div className={styles.header}>
         <h1>My Courses</h1>
+
+        <div className={styles.addCourseBtn}>
+          <Link to={`/users/${userId}/courses/courseForm`}>
+            Add Courses
+          </Link>
+        </div>
```
Comment (jessehartloff) [inline devU-client/src/components/pages/userCoursesListPage.tsx:87]: This says it's a button, but it displays as a plain blue-text link (At least in my browser). Can you make it render as actual button

File: devU-client/src/components/pages/coursesFormPage.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import React,{useState} from 'react'
+
+import { ExpressValidationError } from 'devu-shared-modules'
+
+import PageWrapper from 'components/shared/layouts/pageWrapper'
+
+import RequestService from 'services/request.service'
+
+//import styles from 'components/pages/coursesFormPage.scss'
```
Comment (jessehartloff) [inline devU-client/src/components/pages/coursesFormPage.tsx:9]: Don't leave in commented out code. If it's not going to use, delete it

File: devU-client/src/components/pages/coursesFormPage.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        setLoading(true)
+
+        RequestService.post('/api/courses/', formData)
+            .then(() => {
+                setAlert({ autoDelete: true, type: 'success', message: 'Course Added' })
+            })
+            .catch((err: ExpressValidationError[] | Error) => {
+                const message = Array.isArray(err) ? err.map((e) => `${e.param} ${e.msg}`).join(', ') : err.message
+
+                setAlert({ autoDelete: false, type: 'error', message })
```
Comment (jessehartloff) [inline devU-client/src/components/pages/coursesFormPage.tsx:44]: This alert is too small to contain error messages and it trims it with ellipses and no option to see the full message. Clicking just removes the message. I know you didn't build this feature and it's not part of this PR, but this should be addressed at some point in the future

File: devU-client/src/components/pages/coursesFormPage.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    return (
+        <PageWrapper>
+            <h1>Course Form</h1>
+            <TextField id='name' label='Course Name' onChange={handleChange}/>
+            <TextField id='number' label='Course Number' onChange={handleChange}/>
+            <TextField id='semester' label='Semester' onChange={handleChange}  placeholder='Ex. f2022, w2023, s2024'/>
+            <TextField id='startDate' label='Start Date' onChange={handleChange}/>
+            <TextField id='endDate' label='End Date' onChange={handleChange}/>
```
Comment (jessehartloff) [inline devU-client/src/components/pages/coursesFormPage.tsx:56]: Please use date pickers for these. Manually typing dates in the correct format is nearly unusable

**Review by jessehartloff — CHANGES_REQUESTED** [review]
I was under the impression that the existing code would display all courses and that it only didn't because the DB was empty. I see now that that's not the case. Before moving forward, we'll need a way to see the courses/assignments that have been created

**Comment by kevinzhong930** [comment]
@jessehartloff I believe I pushed all the necessary changes... Hopefully

**Review by jessehartloff — APPROVED** [review]

**Pull request merged**